src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -445,13 +455,17 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
   }
 
   function performDeferredWorkUnsafe(deadline) {
-    if (!nextUnitOfWork) {
+    if (!nextUnitOfWork && !pendingCommit) {
       nextUnitOfWork = findNextUnitOfWork();
     }
-    while (nextUnitOfWork) {
+    while (nextUnitOfWork || pendingCommit) {
```

Comment:
We shouldn't need to read this field at all in a normal cycle (let alone three times). Seems like a smell.